airbyte-bootloader/src/main/java/io/airbyte/bootloader/SecretMigrator.java

```diff
@@ -0,0 +1,200 @@
+            def -> def.getSpec().getConnectionSpecification()));
+
+    final List<DestinationConnection> destinations = configPersistence.listConfigs(ConfigSchema.DESTINATION_CONNECTION, DestinationConnection.class);
+
+    migrateDestinations(destinations, definitionIdToDestinationSpecs);
+  }
+
+  /**
+   * This is migrating the secrets for the source actors
+   */
+  @VisibleForTesting
+  void migrateSources(final List<SourceConnection> sources, final Map<UUID, JsonNode> definitionIdToSourceSpecs) {
+    log.info("Migrating Sources");
+    sources.stream()
+        .map(source -> {
+          final JsonNode migratedConfig = migrateConfiguration(new ConnectorConfiguration(
+              source.getWorkspaceId(),
+              source.getConfiguration(),
+              definitionIdToSourceSpecs.get(source.getSourceDefinitionId())),
+              () -> UUID.randomUUID());
+          source.setConfiguration(migratedConfig);
+          return source;
+        })
+        .forEach(source -> {
+          try {
+            configPersistence.writeConfig(ConfigSchema.SOURCE_CONNECTION, source.getSourceId().toString(), source);
+          } catch (final JsonValidationException | IOException e) {
+            throw new RuntimeException(e);
+          }
+        });
+  }
+
+  /**
+   * This is migrating the secrets for the destination actors
+   */
+  @VisibleForTesting
+  void migrateDestinations(final List<DestinationConnection> destinations, final Map<UUID, JsonNode> definitionIdToDestinationSpecs) {
+    log.info("Migration Destinations");
+
+    destinations.stream().map(destination -> {
+      final JsonNode migratedConfig = migrateConfiguration(new ConnectorConfiguration(
+          destination.getWorkspaceId(),
+          destination.getConfiguration(),
+          definitionIdToDestinationSpecs.get(destination.getDestinationDefinitionId())),
+          () -> UUID.randomUUID());
+      destination.setConfiguration(migratedConfig);
+      return destination;
+    })
+        .forEach(destination -> {
+          try {
+            configPersistence.writeConfig(ConfigSchema.DESTINATION_CONNECTION, destination.getDestinationId().toString(), destination);
+          } catch (final JsonValidationException | IOException e) {
+            throw new RuntimeException(e);
+          }
+        });
+  }
+
+  /**
+   * This is a generic method to migrate an actor configuration
+   * It will extract the secret path form the provided spec and then replace them by coordinates in the actor configuration
+   */
+  @VisibleForTesting
+  JsonNode migrateConfiguration(final ConnectorConfiguration connectorConfiguration, final Supplier<UUID> uuidProvider) {
+    if (connectorConfiguration.getSpecs() == null) {
+      throw new IllegalStateException("No connector definition to match the connector");
+    }
+
+    final AtomicReference<JsonNode> connectorConfigurationJson = new AtomicReference<>(connectorConfiguration.getConfiguration());
```

Comment:
It is because it is accessed in a lambda and expecially since it is set in a lambda, it has the error `Variable used in lambda expression should be final or effectively final` and on of the fix is to use the atomic value even if we are not in a multi thread use case.

Reply:
Got it, makes sense. Thanks